chain/vm/runtime.go

```diff
@@ -55,6 +56,32 @@ type Runtime struct {
 	allowInternal      bool
 }
 
+type safeCBORMarshaler struct {
+	m cbg.CBORMarshaler
+}
+
+func (s *safeCBORMarshaler) MarshalCBOR(w io.Writer) error {
```

Comment:
Q: is is worth wrapping `io.Reader/Writer`s and throw anything that isn't `io.EOF` as fatal? Or can we safely assume that those are always `bytes.Reader` under the hood?

Reply:
probably safe to assume they are all bytes.Readers